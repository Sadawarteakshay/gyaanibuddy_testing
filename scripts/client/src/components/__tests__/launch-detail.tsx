import React from 'react';

import { render, cleanup, screen } from '../../test-utils';
import LaunchDetail from '../launch-detail';
import renderer from 'react-test-renderer';

describe('Launch Detail View', () => {
  // automatically unmount and cleanup DOM after the test is finished.
  afterEach(cleanup);

  it('renders without error', () => {
    render(
      <LaunchDetail
        id={'1'}
        site={'earth'}
        rocket={{ name: 'that one', type: 'big', __typename: 'Rocket', id: '1' }}
      />,
    );
  });

  it('render snapshot', () => {
    const LaunchElement = <LaunchDetail
      id={'1'}
      site={'earth'}
      rocket={{ name: 'that one', type: 'big', __typename: 'Rocket', id: '1' }}
    />;

    render(LaunchElement);

    // Snapshot test - ensure button has no unsuspecting changes between commits
    const ElementTree = renderer
      .create(LaunchElement)
      .toJSON();

    expect(ElementTree).toMatchSnapshot();
  });

  it('should contain correct rocket name and rocket type', () =>{
    const LaunchElement = <LaunchDetail
      id={'1'}
      site={'earth'}
      rocket={{ name: 'that one', type: 'big', __typename: 'Rocket', id: '1' }}
    />;

    render(LaunchElement);

    expect(screen.getByText('earth')).toHaveTextContent('earth'); // TODO: fix this; check the dom or something
  });
});
