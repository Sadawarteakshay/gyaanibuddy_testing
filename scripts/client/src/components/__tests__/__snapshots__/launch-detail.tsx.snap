// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launch Detail View render snapshot 1`] = `
<div
  className="css-1wwf220"
  style={
    Object {
      "backgroundImage": "url(iss.jpg)",
    }
  }
>
  <h3>
    that one
     (
    big
    )
  </h3>
  <h5>
    earth
  </h5>
</div>
`;
